### PR TITLE
✨ Add oracle failsaves in MainRegistry

### PR DESCRIPTION
**Pull request by Thomas-Smets**

### CODE REVIEW

File: src/MainRegistry.sol [hunk 1]
```diff
@@ -140,6 +140,46 @@ contract MainRegistry is IMainRegistry, MainRegistryGuardian {
...
+    /**
+     * @notice Sets a new oracle for the rate baseCurrency-USD.
+     * @param baseCurrency The identifier of the baseCurrency for which the new oracle is set.
+     * @param newOracle The new oracle address.
+     * @dev This function is part of an oracle failsafe mechanism.
+     * New oracles can only be set if the current oracle is not performing as intended:
+     * - A call to the oracle reverts
+     * - The oracle returns the minimum value
+     * - The oracle returns the maximum value
+     * - The oracle didn't upadet for over a week
```
Comment (j-vp) [inline src/MainRegistry.sol:152]:
```suggestion
     * - The oracle didn't update for over a week
```

File: src/MainRegistry.sol [hunk 2]
```diff
@@ -140,6 +140,46 @@ contract MainRegistry is IMainRegistry, MainRegistryGuardian {
...
+     * @dev This function could be called to set an oracle address for the basecurrency USD (since it is initiated with the zero address).
+     * This oracle is however never used, hence would not cause any problems (exept gas waste).
```
Comment (j-vp) [inline src/MainRegistry.sol:154]: should we consider adding a check 'if not basecurrency == 0' then?

Comment (Thomas-Smets) [reply src/MainRegistry.sol:154]: Not really, we never use that oracle, so it would not really matter if a bogus oracle is set

**Review by j-vp — APPROVED** [review]
gtg

**Pull request merged**